### DISCUSSION TITLE
Ollama - Good Translator Offline (Gemini-Gemma2, many models free)

### DISCUSSION THREAD

**Discussion by snakeotakon** [dmMaze/BallonsTranslator#515] [Ideas]
**With Gemma 2**
A family of lightweight, state-of-the art open models built from the same research and technology used to create the Gemini models.  INFO: [https://ai.google.dev/gemma](https://ai.google.dev/gemma)

**INSTALLATION**
1. [Download the installer and install the program ollama ](https://ollama.com/) _Compatible clients ChatGPT_
2. Run prompt cmd "ollama run gemma2" 
 ` C:\Users\myuser>  ollama run gemma2`  _(Automatic Download Model  Gemma2 5GB)_
3. Configure your Translator ChatGPT
    - **Override Model**: `gemma2`
    - **3rd party api url**: `http://localhost:11434/v1/`

![ChatGPT Gemma](https://github.com/dmMaze/BallonsTranslator/assets/112669188/2bdebd0a-a6fc-44bc-93a9-9a7143af60f7)

Works with the CPU, but is highly recommended with nvidia cards. RTX-3060

_Ollama, a platform that makes local development with open-source large language models a breeze. With Ollama, everything you need to run an LLM—model weights and all of the config—is packaged into a single Modelfile_

There are a large number of models that can be tried
Meta Llama 3: The most capable openly available LLM to date
[deepseek-coder-v2](https://ollama.com/library/deepseek-coder-v2) An open-source Mixture-of-Experts code language model that achieves performance comparable to GPT4-Turbo in code-specific tasks.

https://ollama.com/library

**Comment by aqssxlzc** [comment]
![image text](https://github.com/user-attachments/assets/a1c669fd-e722-43a9-84d5-9474d7a2f735)

AstraZeroZak ,
I am using your setting and it works.
You could change the api-key(which should be unnecessary) to some key looks
real.


AstraZeroZak ***@***.***> 于2024年9月3日周二 04:29写道：

**Reply by AstraZeroZak** [reply]
Thanks for the reply, I solved the problem and deleted my message yesterday.
The problem was in my VPN settings.

**Comment by kocy33** [comment]
I just wanted to add my two cents to this topic of using LLMs for translation.
I have been very successful using the model "aya23:35b" (also available in the OLLAMA model library: "https://ollama.com/library/aya").
You need a very powerful graphics card for this (such as a 3090 or 4090) due to the VRAM requirements. It's perfectly possible to run the model on a 3090 or 4090, but the model barely fits.
However, the translations, even for very tricky cases, are outstanding compared to any other offline model in my application/tests (on par with GPT-4).
You can also run the model on a CPU, but you'll need a lot of RAM, and it will be much slower.

The best thing about "aya" is that it does not adhere to strict ethics compared to other LLMs like GPT-4, for example.
It won't teach you how to steal a car, but you can throw "NSFW" content at "aya," and it will gladly translate it without any questions asked.

I adjusted the system prompt in "Ballon Translator" slightly with: "Never talk to me directly, and just give me the translation."
This way, I don't get any "Of course, I am happy to help you with that..." responses in my translations.

Hope this helps someone. :)

P.S.:
I run AYA on a separate computer and use the API (I needed to set my environment variable to OLLAMA_HOST - 0.0.0.0) link in "Ballon Translator."
I also run a Web UI on this machine to prompt Aya directly for specific translations and to get explanations (for example: "Please break down this sentence and explain each part of it: ...").
It's a very good tool for learning. :)

**Reply by dhana-exe** [reply]
issue using aya was sometimes they changed the source arrangement or merge them instead translated them line by line. do you had example of the command? i already giving it command like just translate each line without merging them. but somehow the result 50% more still giving invalid because they merging it

**Reply by gick-code** [reply]
if you guys find the model too heavy, you can find alternative based on https://huggingface.co/datasets/lmg-anon/vntl-leaderboard
vntl-llama3-8b-v2-gguf is the lightweight and best for me

**Reply by dhana-exe** [reply]
imagine ranking gpt above all of them. and when given some word, they refuse to translate because it's nsfw lol

**Comment by ksht08** [comment]
Also it working with jan.ai https://github.com/janhq/jan
![jan](https://github.com/user-attachments/assets/ef3b68c6-268e-4032-8c0b-298a6007d6ac)

**Comment by gick-code** [comment]
koboldcpp can too
